apps/alerting/notifications/apis/resource/receiver/v0alpha1/ext.go

```diff
@@ -0,0 +1,75 @@
+package v0alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apiserver/pkg/registry/generic"
+
+	"github.com/grafana/grafana/apps/alerting/common"
+)
+
+func (o *Receiver) GetProvenanceStatus() string {
+	if o == nil || o.Annotations == nil {
+		return ""
+	}
+	s, ok := o.Annotations[common.ProvenanceStatusAnnotationKey]
+	if !ok || s == "" {
+		return common.ProvenanceStatusNone
+	}
+	return s
+}
+
+func (o *Receiver) SetProvenanceStatus(status string) {
+	if o.Annotations == nil {
+		o.Annotations = make(map[string]string, 1)
+	}
+	if status == "" {
+		status = common.ProvenanceStatusNone
+	}
+	o.Annotations[common.ProvenanceStatusAnnotationKey] = status
+}
+
+func (o *Receiver) SetAccessControl(action string) {
+	if o.Annotations == nil {
+		o.Annotations = make(map[string]string, 1)
+	}
+	o.Annotations[AccessControlAnnotation(action)] = "true"
+}
+
+// AccessControlAnnotation returns the key for the access control annotation for the given action.
+// Ex. grafana.com/access/canDelete.
+func AccessControlAnnotation(action string) string {
+	return fmt.Sprintf("%s%s/%s", common.InternalPrefix, "access", action)
+}
+
+func (o *Receiver) SetInUse(routesCnt int, rules []string) {
+	if o.Annotations == nil {
+		o.Annotations = make(map[string]string, 2)
+	}
+	o.Annotations[InUseAnnotation("routes")] = fmt.Sprintf("%d", routesCnt)
+	o.Annotations[InUseAnnotation("rules")] = fmt.Sprintf("%d", len(rules))
+}
+
+// InUseAnnotation returns the key for the in-use annotation for the given resource.
+// Ex. grafana.com/inUse/routes, grafana.com/inUse/rules.
+func InUseAnnotation(resource string) string {
+	return fmt.Sprintf("%s%s/%s", common.InternalPrefix, "inUse", resource)
+}
+
+func SelectableFields(obj *Receiver) fields.Set {
+	if obj == nil {
+		return nil
+	}
+	set := fields.Set{
+		"metadata.provenance": obj.GetProvenanceStatus(),
```

Comment:
@yuri-tceretian i think i've only seen selectable fields exposed using paths that match the location the stored object. are you basing the annotation remapping on an example you've seen?

Reply:
We're not clear about where to store this field yet but the UI wants to filter by this status. Perhaps it can live in status or even a separate resource. Maybe labels. Also we plan to enable this field for writing and check for additional permissions. 
I would appreciate ideas about the better place for it.

Reply:
I would consider using a label instead, unless you need more structure / stronger typing around it, in which case I'd move this to `.status.provenance` or something like that.

But I definitely wouldn't want to add a field that Kubernetes doesn't support.

Reply:
@JacobsonMT and I discussed this and decided to remove the selectors because they are not used in UI anyway.